src/form-engine.component.tsx

```diff
@@ -84,6 +85,7 @@ const FormEngine: React.FC<FormProps> = ({
   markFormAsDirty,
 }) => {
   const session = useSession();
+  const { encounterRole } = useEncounterRole();
```

Comment:
Help me understand the reason behind moving this hook call here.

Reply:
I need to pass the encounterRole to the EncounterForm, I tried having it in the encounter form but it takes long to initialize and the encounter context initializes with `undefined` role, having it in the parent wrapper helps.

Reply:
I think this issue should have been addressed after the discussion, right?

Reply:
I am not sure because this was happening even before the view mode but I can double check it and see.

Reply:
>I need to pass the encounterRole to the EncounterForm, I tried having it in the encounter form but it takes long to initialize and the encounter context initializes with undefined role, having it in the parent wrapper helps.

I understand the issue and it is something I addressed as part of this [commit](https://github.com/openmrs/openmrs-form-engine-lib/pull/273/commits/331d319b56b21a57aa751010019e9a89337b499e). @hadijahkyampeire This should live in the encounter form component.